samples/base/Logger.cpp

```diff
@@ -60,9 +64,13 @@ namespace darwin {
             fmt << "\"filter\":\"" << _name << "\",";
             fmt << "\"message\":\"" << logMsg << "\"";
             fmt << '}';
-            _fileMutex.lock();
-            _file << fmt.str().c_str() << std::endl;
-            _fileMutex.unlock();
+            if (not _file.is_open() and not this->openLogFile()) {
+                std::clog << fmt.str() << std::endl;
+            } else {
+                _fileMutex.lock();
+                _file << fmt.str().c_str() << std::endl;
+                _fileMutex.unlock();
```

Comment:
Maybe at some point, we will have to queue log messages? having this mutex shared among all threads may slow things if a filter use the logger extensively.

Reply:
maybe, I believe the current stand is to offload queuing/buffering/logging performances to the underlying OS specifics, but I agree the logging module could certainly be revamped (possibly with an external library)

Reply:
change made, I leave you to comment/resolve the conversation